python/lsst/dc2gen3/patch_dc2.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python
+"""Modify a DESC DC2 to prepare it for conversion to gen3
+"""
+
+# imports
+
+import argparse
+import sys
+import os
+import shutil
+
+from astropy.io import fits
+import lsst.log
+import lsst.log.utils
+
+from lsst.daf.persistence import Policy
+from lsst.obs.base.gen2to3.repoWalker import PathElementParser
+
+# constants
+
+DATATYPES_TO_CONVERT = ("flat", "sky", "SKY")
+POLICY_FILE = Policy.defaultPolicyFile(
+    "obs_lsst", "lsstCamMapper.yaml", "policy"
+)
+FIX_HEADER = False
+
+# exception classes
+
+# interface functions
+
+
+def copy_files(origin_repo, new_repo, policy_file=POLICY_FILE):
+    """Copy files from a DESC DC2 gen2 origin repo to one with modified names
+    to change parsed filter names to match that expected by the gen3 butler.
+
+    Parameters
+    ==========
+    origin_repo : str
+        full path to origin gen2 (POSIX) filestore
+    new_repo : str
+        full path to destination (POSIX) filestore
+    policy_file : str
+        Policy file from which to load templates
+
+    """
+    log = lsst.log.Log.getLogger("convertRepo")
+    policy = Policy(policy_file)
+    for data_type in DATATYPES_TO_CONVERT:
+        if data_type == "SKY":
+            template = policy["calibrations"]["sky"]["template"]
+            template = template.replace("sky", "SKY")
+        else:
+            template = policy["calibrations"][data_type]["template"]
+
+        for old_path, new_path in _transform_pairs(
+            origin_repo, new_repo, template
+        ):
+
+            if FIX_HEADER:
+                assert False, "This path is untested"
+                with fits.open(old_path) as hdulist:
+                    for hdu in hdulist:
+                        if "FILTER" in hdu.header:
+                            old_filter = hdu.header["FILTER"]
+                            new_filter = _transform_filter_name(old_filter)
+                            hdu.header["FILTER"] = new_filter
+                    hdulist.writeto(new_path)
+            else:
+                log.info("Copying %s to %s", old_path, new_path)
```

Comment:
If you use `lsst.daf.butler.ButlerURI` you can make use of the `ButlerURI.transfer_from()` method and then support all the standard `transfer` modes.